app/src/main/java/com/example/reduxtestapp/ui/country/CountryScreen.kt

```diff
@@ -184,8 +184,9 @@ private fun SearchBar(
     onSendSearchQuery: (String) -> Unit,
     modifier: Modifier = Modifier
 ) {
+
     LaunchedEffect(key1 = searchQuery) {
-        delay(1000L)
+        if (searchQuery.isNotEmpty()) delay(1000L)
         onSendSearchQuery(searchQuery)
     }
```

Comment:
this will still cause an extra network call on first load, so maybe put `onSendSearchQuery(searchQuery)` inside the if too?